### DISCUSSION TITLE
Referencing past calls

### DISCUSSION THREAD

**Discussion by IcedQuinn** [deeprjs/deepr#11] [Q&A] [answered]
Was talking to someone about [JMAP](https://jmap.io/spec-core.html#references-to-previous-method-results) recently which contains a system to reference the result of a previous call.

It works like basically putting one or more calls in flight that assign their return values to some variable `v` and then a third call is able to say one of its parameters is whatever was returned and placed in to `v`. There is then a piece of the reference that specifies which particular subset of that result is the value being picked.

It looks like this kind of exists here as "source value" but the docs are very sparse as to why you would use them.

Are these basically the same thing, or is that kind of restructuring not really supported in deepr?

**Comment by mvila** [comment, marked as answer]
The purpose of source values is different than what you think. Basically, it allows specifying an object and then executing some methods on it. The way the object is serialized/deserialized is up to the developer and outside of the scope of Deepr.

**Reply by IcedQuinn** [reply]
So there is no way to chain calls, outside of while inside the {..scope..} of a particular graph node at the time.

**Reply by mvila** [reply]
That's right.
